ui/desktop/electron-app/config/cli.js

```diff
@@ -69,12 +69,10 @@ const downloadArtifact = (version) => {
 const extract = (artifactPath, destination) => {
   if (!fs.existsSync(destination)) fs.mkdirSync(destination);
 
-  return fs
-    .createReadStream(artifactPath)
-    .pipe(unzipper.Extract({ path: destination }))
-    .on('close', () => {
-      console.log('Extract artifact to: ', destination);
-    });
+  const zip = new admZip(artifactPath);
+  // The first boolean flag is to overwrite files, the second is to preserve original
+  // file permissions which doesn't seem to be documented.
+  return zip.extractAllTo(destination, true, true);
```

Comment:
It's not documented but [this method](https://github.com/cthackers/adm-zip/blob/05c25b123284cd0d3c8930995914ed09dbb8cf43/adm-zip.js#L598) has a boolean flag that lets us preserve the original permissions 🤷